src/sync/installation.ts

```diff
@@ -406,22 +417,36 @@ export const processInstallation =
 				}
 
 				// Continue sync when a 404/NOT_FOUND is returned
-				if (handleNotFoundErrors(err, job, nextTask)) {
+				if (isNotFoundError(err, job, nextTask)) {
 					const edgesLeft = []; // No edges left to process since the repository doesn't exist
 					await updateJobStatus(queues, job, edgesLeft, task, repositoryId);
 					return;
 				}
 
-				await subscription.update({ syncStatus: "FAILED" });
+				if (await booleanFlag(BooleanFlags.CONTINUE_SYNC_ON_ERROR, false, jiraHost)) {
+
+					logger.warn({ job, task: nextTask, err }, "Task failed, continuing with next task");
 
-				const host = subscription.jiraHost || "none";
-				logger.warn({ job, task: nextTask, err, jiraHost: host }, "Sync failed");
+					// marking the current task as failed
+					await subscription.updateRepoSyncStateItem(nextTask.repositoryId, getStatusKey(nextTask.task as TaskType), "failed");
```

Comment:
/nit it would be great to have an enum with the task statuses rather than use strings. However not sure if it worth changing if we want to rewrite it anyway

Reply:
Yeah. There is a `TaskStatus` type defined as `"complete" | "pending" | "failed"`, but we should probably migrate that to a proper enum some time.